api/nodes/serializers.py

```diff
@@ -130,6 +131,125 @@ def update(self, instance, validated_data):
+    family_name = ser.CharField(read_only=True, help_text='For bibliographic citations')
+    suffix = ser.CharField(read_only=True, help_text='For bibliographic citations')
+    date_registered = ser.DateTimeField(read_only=True)
+    gravatar_url = ser.CharField(read_only=True, help_text='URL for the icon used to identify the user. Relies on '
+                                                           'http://gravatar.com ')
+    employment_institutions = ser.ListField(read_only=True, source='jobs', help_text='An array of dictionaries '
+                                                                                     'representing the '
+                                                                                     'places the user has worked')
+    educational_institutions = ser.ListField(read_only=True, source='schools', help_text='An array of dictionaries '
+                                                                                         'representing the places the '
+                                                                                         'user has attended school')
+    social_accounts = ser.DictField(read_only=True, source='social', help_text='A dictionary of various social media '
+                                                                               'account identifiers including an array '
+                                                                               'of user-defined URLs')
+    links = LinksField({
+        'html': 'absolute_url',
+        'contributor-self': Link('nodes:node-contributor-detail', kwargs={'user_id': '<_id>', 'node_id': '<node_id>'}),
+        'nodes': {
+            'relation': Link('users:user-nodes', kwargs={'user_id': '<_id>'}),
+        },
+    })
+
+    def absolute_url(self, obj):
+        return obj.absolute_url
+
+    def create(self, validated_data):
+        current_user = self.context['request'].user
+        auth = Auth(current_user)
+        node = self.context['view'].get_node()
+        user = User.load(validated_data['_id'])
+        if not user:
+            raise NotFound('User with id {} cannot be found.'.format(validated_data['_id']))
+        elif user in node.contributors:
+            raise ValidationError('User {} already is a contributor.'.format(user.username))
+        bibliographic = validated_data['bibliographic'] == "True"
+        permission_field = validated_data['permission']
+        node.add_contributor(contributor=user, auth=auth, save=True)
+        self.set_visibility(bibliographic, user, node)
+        self.set_permissions(permission_field, user, node)
+        user.node_id = node._id
+        user.bibliographic = node.get_visible(user)
+        return user
+
+    def set_visibility(self, bibliographic, user, node, is_admin=True, is_current=False):
+        if is_admin and bibliographic:
+            node.set_visible(user, True, save=True)
```

Comment:
And chances are you don't have to do anything to use it. See if the error bubbles up properly if you just try to set the visibility under the improper circumstances.